engine/src/main/java/org/terasology/utilities/download/MultiFileDownloader.java

```diff
@@ -26,56 +27,75 @@
 import java.util.Map;
 import java.util.concurrent.Callable;
 
+/**
+ * Downloads multiple files.
+ */
+
 public class MultiFileDownloader implements Callable<List<Path>> {
-    private Map<URL, Path> urlToTargetMap;
+    private Map<URI, Path> uriToTargetMap;
     private MultiFileTransferProgressListener progressListener;
 
-    public MultiFileDownloader(Map<URL, Path> urlToTargetMap, MultiFileTransferProgressListener progressListener) {
-        this.urlToTargetMap = urlToTargetMap;
+    /**
+     * @param uriToTargetMap
+     * @param progressListener a progress listener. Will be called with 0 repeatedly if size cannot be determined.
+     */
+    public MultiFileDownloader(Map<URI, Path> uriToTargetMap, MultiFileTransferProgressListener progressListener) {
+        this.uriToTargetMap = uriToTargetMap;
         this.progressListener = progressListener;
     }
 
     @Override
     public List<Path> call() throws IOException {
 
         // Obtain the size of each file
-        final int numFiles = urlToTargetMap.size();
+        final int numFiles = uriToTargetMap.size();
         if (numFiles < 1) { // avoid reporting progress like "Retrieving file size 1 of 0"
             return Collections.emptyList();
         }
-        final Map<URL, Long> fileSizes = getSizes();
+        final Map<URI, Long> fileSizes = getSizes();
         final long totalBytes = sum(fileSizes.values());
 
         List<Path> downloadedFiles = new ArrayList<>();
         int index = 0;
         long completedFilesBytes = 0;
-        for (Map.Entry<URL, Path> entry : urlToTargetMap.entrySet()) {
+        for (Map.Entry<URI, Path> entry : uriToTargetMap.entrySet()) {
             final int downloadedFileCount = index + 1;
             final long previousFilesBytes = completedFilesBytes;
             SingleFileTransferProgressListener singleDownloadListener = (fileDownloadedBytes, fileTotalBytes) ->
-                progressListener.onDownloadProgress(previousFilesBytes + fileDownloadedBytes, totalBytes, downloadedFileCount, numFiles);
-            SingleFileDownloader fileDownloader = new SingleFileDownloader(entry.getKey(), entry.getValue(),
+                    progressListener.onDownloadProgress(previousFilesBytes + fileDownloadedBytes, totalBytes, downloadedFileCount, numFiles);
+
+            // Converting URI to URL in order to pass it SingleFileDownloader's constructor
+            URL url = entry.getKey().toURL();
+            SingleFileDownloader fileDownloader = new SingleFileDownloader(url, entry.getValue(),
                     singleDownloadListener);
             downloadedFiles.add(fileDownloader.call());
             completedFilesBytes += fileSizes.get(entry.getKey());
             index++;
         }
         return downloadedFiles;
     }
-
-    private Map<URL, Long> getSizes() throws IOException {
-        Map<URL, Long> result = new HashMap<>();
+    /**
+     * This method returns a Hash Map with the the uri as a kay and the size as a value.
```

Comment:
Your javadoc is just repeating the method signature. You shouldn't add javadocs unless you have new / useful information.